### PR TITLE
feat: add icons package

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**
Add icons to the library.

To be used as
```tsx
import { GitHubIcon, SearchIcon } from 'bdp-ui/icons';
```

### CODE REVIEW

File: src/icons/FilterCloseIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  <svg
+    width={width}
+    viewBox="0 0 20 20"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+    {...props}
+  >
+    <path
+      d="M9.93317 16.6L3.33317 10L9.93317 3.4L11.1082 4.58333L6.52484 9.16666H18.3332V10.8333H6.52484L11.1165 15.4167L9.93317 16.6ZM3.33317 10V1.66666H1.6665V18.3333H3.33317V10Z"
+      // strokeWidth="4"
```
Comment (IgboPharaoh) [inline src/icons/FilterCloseIcon.tsx:14]: We can remove commented code here

**Review by IgboPharaoh — COMMENTED** [review]
Code looks good to me, just a few comments

**Comment by Emmanuel-Develops** [comment]
Storybook preview to view all Icons

Screenshot 2024-08-09 at 2 58 38 PM

File: src/icons/stories/icon.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      }
+      return (<IconItem key={name} IconComponent={IconComponent} name={name} />);
+    })}
+  </IconShowcase>
+);
+
+export const AllIcons: Story = {
+  args: {
+    width: 24,
+    // color: "black",
```
Comment (Extheoisah) [inline src/icons/stories/icon.stories.tsx:49]: unused comment?!

File: src/icons/stories/icon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    className: "text-orange-500",
+    pathProps: {
+      strokeWidth: 1.5,
+    }
+  },
+  render: (args) => <IconGrid {...args} />,
+};
+
+// const Template: StoryFn<ButtonProps> = (args) => <Button {...args} />;
```
Comment (Extheoisah) [inline src/icons/stories/icon.stories.tsx:58]: unused comment from line 58 to 70

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**